src/build/get_supported_os_target.mjs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import xml2js from 'xml2js';
+import fs from 'fs/promises';
+
+/*
+  Inputs:
+  => platform: the platform we have a baseline level of support for
+
+  Outputs:
+  => number representing the lowest version of that platform we support
+*/
+export async function getSupportedOSTarget(platform) {
```

Comment:
Different OSes have different concepts. 

I'd create different function for different OSes.
You don't gain much by using `getSupportedOSTarget('ios')` vs `getIosDeploymentTarget()`. In fact, the second is better, because it's more precise. It specifies exactly what the number is.

Same for `getMacosDeploymentTarget()`.

Reply:
Also, do we use the numbers for Android, Linux and Windows? That's probably dead code.